### ISSUE TITLE
Compile error when using "experimental" feature

### ISSUE THREAD

**Issue by fkohlgrueber** [esp-rs/embedded-svc#2]
On current nightly, compiling the crate with the "experimental" feature fails:
```
cargo build --features experimental
```

```
error: Missing required bounds on Request
 --> src/http/client.rs:8:5
  |
8 |     type Request<'a>: HttpRequest<'a, Error = Self::Error>;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
  |                                                           |
  |                                                           help: add the required where clauses: `where Self: 'a`

error: Missing required bounds on Slot
  --> src/ota.rs:70:5
   |
70 |     type Slot<'a>: OtaSlot;
   |     ^^^^^^^^^^^^^^^^^^^^^^-
   |                           |
   |                           help: add the required where clauses: `where Self: 'a`

error: Missing required bounds on Update
  --> src/ota.rs:71:5
   |
71 |     type Update<'a>: OtaUpdate;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^-
   |                               |
   |                               help: add the required where clauses: `where Self: 'a`

error: Missing required bounds on OtaRead
   --> src/ota.rs:133:5
    |
133 |     type OtaRead<'a>: OtaRead<Error = Self::Error>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
    |                                                   |
    |                                                   help: add the required where clauses: `where Self: 'a`
```

Not sure when this was introduced, but it seems like CI doesn't have coverage for the "experimental" feature.

**Comment by fkohlgrueber** [comment]
Seems like it's related to recent changes in nightly. See for example here: https://github.com/rust-lang/rust/issues/90808

**Comment by ivmarkov** [comment]
Fixed with 0.13.
NOTE: The experimental features are WIP and not even tested (that is, aside from the HTTP client).
